### PR TITLE
[Feature]: Add in-app plugin repo browser & management

### PR DESCRIPTION
**Pull request by sethwv**
## Description

Adds an in-app plugin repository browser so users can discover, install, and update plugins from remote repos without manually downloading zips.

**Backend**

- New `PluginRepo` model storing repo URL, cached manifest, GPG public key, and signature verification status, with three migrations (create model, add signing fields, add managed-plugin FK/slug on `PluginConfig`)
- Data migration seeds the official Dispatcharr repo on first run
- API endpoints for repo CRUD, manifest refresh, available-plugin aggregation, per-plugin detail manifest fetch, install-from-repo, and repo refresh settings
- GPG signature verification using detached signatures over canonical JSON (via `python-gnupg`)
- SHA256 integrity check on downloaded plugin zips (optional -- only validated when the manifest provides a hash)
- `registry_name` is required in repo manifests; names that look official (e.g. containing "official", "dispatcharr repo") are rejected for third-party repos
- Celery periodic task for background manifest refresh (default 6h, configurable or disabled)
- `_unmanage_dropped_slugs` clears the managed link on installed plugins whose slug disappears from a refreshed manifest
- `PluginConfig` gains `source_repo` FK and `slug` so installed plugins can be tracked back to a repo for update detection

**Frontend**

- New `/plugins/browse` page (`PluginBrowse.jsx`) with search, repo-source filtering, and pagination
- `AvailablePluginCard.jsx` handles install, update, version history, and detail modal (supports both per-plugin manifests and synthesized detail for repos without one)
- `PluginCard.jsx` now shows Managed/Unmanaged/Update Available badges with clickable detail view
- Zustand `plugins` store extended with repo management and install actions
- `api.js` gains all plugin-repo API methods
- Navigation split into "My Plugins" and "Find Plugins" sub-items
- Manual import of a repo-managed plugin shows a warning that it breaks repo management

## Screenshots
image
image
image
image
image
image

There's a bunch more to see, but I think this is enough screenshot spam.

## How was it tested?

- Manual testing against a local dev instance (Docker Compose + Vite dev server) with the official Dispatcharr plugin repo and a test third-party repo
- Verified full lifecycle: add repo, browse available plugins, install, enable, update, delete, re-install
- Tested GPG signature verification with both valid and invalid signatures
- Tested SHA256 mismatch rejection
- Tested registry_name validation (missing name, official-sounding names on non-official repos)
- Tested manifest refresh (manual, auto-fetch on first browse, periodic via Celery beat)
- Tested slug-drop unmanagement when a plugin is removed from a repo manifest
- Tested manual zip import over a managed plugin (breaks link, shows warning)
- Tested install flow edge cases: version compatibility checks, download size limits, already-installed detection, different-repo conflict display
- Existing CI tests pass (frontend vitest suite, backend tests)

## Checklist

Please check every item before marking this PR as ready for review. Unchecked items will be flagged during review.

- [x] I have read the [CONTRIBUTING.md](../blob/dev/CONTRIBUTING.md) in full
- [x] I agree to the [Contributor License Agreement](../blob/dev/CONTRIBUTING.md#contributor-license-agreement)
- [x] I understand — line by line — every change in this PR and can explain it if asked
- [x] This PR targets the `dev` branch
- [x] Backend: migrations are included if any models were changed
- [x] Backend: new API endpoints appear correctly in the OpenAPI schema
- [x] Frontend: ESLint and Prettier pass cleanly (`npm run lint`, `npm run format`)
- [x] Tests are included for new functionality
- [x] Existing tests still pass
- [x] No `console.log`, `print()`, debug statements, or commented-out code is left in the diff
- [x] I have not reformatted or refactored code outside the scope of this change

### CODE REVIEW

**Comment by sethwv** [comment]
Squashed commits at the request of @SergeantPanda

File: frontend/src/components/cards/AvailablePluginCard.jsx [hunk 1]
```diff
@@ -0,0 +1,967 @@
...
+                        leftSection={installing ? <Loader size={14} /> : <Download size={14} />}
+                        disabled={!selCompatible || installing}
+                        onClick={() => {
+                          if (!selectedVersionData?.url) return;
+                          doInstall({
+                            repo_id: plugin.repo_id,
+                            slug: plugin.slug,
+                            version: selectedVersion,
+                            download_url: selectedVersionData.url,
+                            sha256: selectedVersionData.checksum_sha256,
```
[the remark this answers is not in the document]
Comment (sethwv) [reply frontend/src/components/cards/AvailablePluginCard.jsx]: The spec will need to be updated, checksum_sha256 is the correct field for per-plugin manifests.

**Comment by SergeantPanda** [comment]
Looks REALLY good, thank you again for all of the time you spent on this.


Well thought out and executed.
